pkg/git/provider_gitlab.go

```diff
@@ -94,6 +112,36 @@ func (g *gitlab) checkApiScope(ctx context.Context, token string) error {
 	return nil
 }
 
+func (g *gitlab) checkTokenType(token string, ctx context.Context) (string, error) {
+	userRes, err := g.request(ctx, token, http.MethodGet, "user")
+
+	if err != nil {
+		return "", fmt.Errorf("failed getting user: %w", err)
+	}
+
+	bodyBytes, err := io.ReadAll(userRes.Body)
+	if err != nil {
+		return "", fmt.Errorf("failed reading user body: %w", err)
+	}
+
+	var body useResBody
+	err = json.Unmarshal(bodyBytes, &body)
+	if err != nil {
+		return "", fmt.Errorf("failed parse user body: %w", err)
+	}
+	if body.Bot {
+		if strings.HasPrefix(body.Username, "project") {
+			return "project", nil
+		}
+		return "group", nil
+	}
+
+	defer userRes.Body.Close()
```

Comment:
Should be moved to line 121, right after err is checked. otherwise you will have a leak